numpy/lib/format.py

```diff
@@ -549,7 +549,7 @@ def _filter_header(s):
 
     tokens = []
     last_token_was_number = False
-    # adding newline as python 2.7.5 workaround
+    # Add newline as EOF workaround
```

Comment:
not adding the "\n" on Python 3 results in a large number of test errors.
I didn't investigate.

Reply:
There's a comment on line 565 that's also relevant

Reply:
Yep that fixes it, I feel blind not to have seen that 10 lines away.